Source/Core/Core/AchievementManager.cpp

```diff
@@ -131,15 +131,52 @@ void AchievementManager::LoadGameByFilenameAsync(const std::string& iso_path,
 
     const auto fetch_game_data_response = FetchGameData();
     m_is_game_loaded = fetch_game_data_response == ResponseType::SUCCESS;
+
+    // Claim the lock, then queue the fetch unlock data calls, then initialize the unlock map in
+    // ActivateDeactiveAchievements. This allows the calls to process while initializing the
+    // unlock map but then forces them to wait until it's initialized before making modifications to
+    // it.
+    m_lock.lock();
+    LoadUnlockData([](ResponseType r_type) {});
+    ActivateDeactivateAchievements();
+    m_lock.unlock();
     callback(fetch_game_data_response);
   });
 }
 
+void AchievementManager::LoadUnlockData(const ResponseCallback& callback)
+{
+  m_queue.EmplaceItem([this, callback] {
+    const auto hardcore_unlock_response = FetchUnlockData(true);
+    if (hardcore_unlock_response != ResponseType::SUCCESS)
+    {
+      callback(hardcore_unlock_response);
+      return;
+    }
+
+    callback(FetchUnlockData(false));
+  });
+}
+
+void AchievementManager::ActivateDeactivateAchievements()
+{
+  bool enabled = Config::Get(Config::RA_ACHIEVEMENTS_ENABLED);
+  bool unofficial = Config::Get(Config::RA_UNOFFICIAL_ENABLED);
+  bool encore = Config::Get(Config::RA_ENCORE_ENABLED);
+  for (u32 ix = 0; ix < m_game_data.num_achievements; ix++)
+  {
+    auto iter =
+        m_unlock_map.insert({m_game_data.achievements[ix].id, UnlockStatus{.game_data_index = ix}});
+    ActivateDeactivateAchievement(iter.first->first, enabled, unofficial, encore);
+  }
+}
```

Comment:
Should this clear the map before inserting new entries?

Reply:
No - because the unlock map should be cleared when the game is closed, which I forgot to do and will add instead. 😅 That said, I don't want to eliminate the possibility that ActivateDeactivateAchievement could be called for other reasons than first-time elimination.